File: 3325-find-the-largest-area-of-square-inside-two-rectangles/find-the-largest-area-of-square-inside-two-rectangles.java
class Solution {
    public long largestSquareArea(int[][] bottomLeft, int[][] topRight) {
        int n = bottomLeft.length;
        long maxArea = 0;

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int x1 = Math.max(bottomLeft[i][0], bottomLeft[j][0]);
                int y1 = Math.max(bottomLeft[i][1], bottomLeft[j][1]);
                int x2 = Math.min(topRight[i][0], topRight[j][0]);
                int y2 = Math.min(topRight[i][1], topRight[j][1]);

                if (x1 < x2 && y1 < y2) {
                    int width = x2 - x1;
                    int height = y2 - y1;
                    int side = Math.min(width, height);
                    long area = (long) side * side;
                    maxArea = Math.max(maxArea, area);
                }
            }
        }

        return maxArea;
    }
}

File: 3325-find-the-largest-area-of-square-inside-two-rectangles/find-the-largest-area-of-square-inside-two-rectangles.py
class Solution:
    def largestSquareArea(self, bottomLeft, topRight):
        n = len(bottomLeft)
        max_area = 0

        for i in range(n):
            for j in range(i + 1, n):
                x1 = max(bottomLeft[i][0], bottomLeft[j][0])
                y1 = max(bottomLeft[i][1], bottomLeft[j][1])
                x2 = min(topRight[i][0], topRight[j][0])
                y2 = min(topRight[i][1], topRight[j][1])

                if x1 < x2 and y1 < y2:
                    width = x2 - x1
                    height = y2 - y1
                    side = min(width, height)
                    area = side * side
                    max_area = max(max_area, area)

        return max_area
